File: backend/models/scaffold.py
from typing import Optional
from pydantic import BaseModel


class ClozeItem(BaseModel):
    item_id: str
    # Step 1 fields
    sentence_skeleton: Optional[str] = None
    correct_facts: Optional[list[str]] = None
    blank_labels: Optional[list[str]] = None
    source_sentence: Optional[str] = None
    # Step 2 fields
    given_facts: Optional[str] = None
    sentence_with_blanks: Optional[str] = None
    chinese_prompt: Optional[str] = None
    correct_structure: Optional[list[str]] = None
    phrasebank_source: Optional[str] = None


class Step1Response(BaseModel):
    session_id: str
    items: list[ClozeItem]


class Step2Response(BaseModel):
    session_id: str
    items: list[ClozeItem]


class Step3Response(BaseModel):
    session_id: str
    chinese_outline: list[str]

File: frontend/src/types/scaffold.ts
export interface ClozeItem {
  item_id: string
  // Step 1
  sentence_skeleton?: string
  correct_facts?: string[]
  blank_labels?: string[]
  source_sentence?: string
  // Step 2
  given_facts?: string
  sentence_with_blanks?: string
  chinese_prompt?: string
  correct_structure?: string[]
  phrasebank_source?: string
}

export interface Step1Response {
  session_id: string
  items: ClozeItem[]
}

export interface Step2Response {
  session_id: string
  items: ClozeItem[]
}

export interface Step3Response {
  session_id: string
  chinese_outline: string[]
}

export type ScaffoldStep = 1 | 2 | 3

export type StepStatus = 'idle' | 'loading' | 'active' | 'submitted' | 'complete'
